### ISSUE TITLE
v1.5.12 is the copyright placement on illustrations (for David C. Cook) supposed to work now?

### ISSUE THREAD

**Issue by davidc86** [sillsdev/ptx2pdf#397]
I can't get the copyright placement feature on DCC illustrations to work.
Also, since the DCC copyright is required on all illustrations, do we have to manually mark each one of these as needing this? Could PTXprint just handle that? And we only adjust where it shows?

**Comment by markpenny** [comment]
Yes, we've taken a few baby steps in this direction, and hope to make it much easier for DCC copyright to be applied automatically to all CO and CN illustrations. We may end up building a shared database of WHERE to optimally place the text on each image (and then allow the user to tweak that position if needed).


> I can't get the copyright placement feature on DCC illustrations to work

Does nothing come out at all? Or are you unable to control the position? Or the shading? Are you using the English UI (or the Indonesian?) - I have a feeling it won't (yet) work with the non-English UI.

**Comment by davidc86** [comment]
Currently I'm only using English UI (minimize variables).
On the Picture Details tab (nice tweak that, btw), I see at the bottom Copyright Attribution: is a nice green "Automatic".
Looking at the Credits Overlay tab, it is **_unclear_** if I'm supposed to enter anything in the Image Credit Text.
Hmm.... now it is showing a gray "(c) David C. Cook" (which is what I was expecting, but it was not showing before). Ah! The problem is the cursor defaults to that text box, which opens up empty! This is confusing. Because then it looks like I need to add that for each DCC illustration! Yikes. So maybe the cursor should NOT default into that text box, so then the user can see the grayed out attribution already there?  

When that text box is not in focus, this is what I'm seeing in PTXprint: 
![image](https://user-images.githubusercontent.com/65363206/108015014-3904bf00-6fdd-11eb-8ecc-07586cc784b2.png)

And this is what I'm seeing in the PDF:
![image](https://user-images.githubusercontent.com/65363206/108015042-48840800-6fdd-11eb-8359-ce16d170103c.png)

So basically I'm not seeing anything. Am I supposed to pick a background box? I thought "None" was the best choice (which I assume means "transparent", i.e. the "text box" won't create a square cutout in the illustration if it overlaps slightly). Unless "None" means "Don't show it" -- in which case this UI needs tweaking. It looks as though it WILL do it. (I don't see how to tell not to.)  

But just to verify, I ran this same print process 3 times, using None, Light, and Dark backgrounds and none showed. 

Also, the "Edit Style" hyperlink should be active correct? It would be nice if that style were included standard in the PTXprint installer.

**Comment by markpenny** [comment]
Ah - now I understand the confusion...
The gray "(c) David C. Cook" is only placeholder text (hinting to the user that they actually have to enter something in that text box). But because it is exactly what you want, you haven't realized that it isn't actually there...

So yes, I'll change the UI to make it more obvious and/or auto-fill the box with the David C Cook line or DCC if we find a CO or CN image.

Note that you can just type DCC and it expands it for you..

**Comment by markpenny** [comment]
For now I have added a Tooltip and instructions to enter something into that box. Later we may do some magic to auto-populate the field:
![image](https://user-images.githubusercontent.com/12962708/108043628-30c48680-70a6-11eb-8651-fe06a56801e8.png)

**Comment by davidc86** [comment]
Nice tool tip. 
You mention "Later we **_may_** do some magic..."
But the Settings tab has this marked as "Automatic"... so that would imply it already does it, right? (Or maybe this is just referring to the colophon?)

**Comment by markpenny** [comment]
Yes, the Details tab says "Automatic" for the Image Credits that turn up in the Colophon. But for the Credit Overlays (which AFAIK) are only required for DCC images, you still have to [currently] fill in the field. I'd like to just make that happen automatically at run time - so that the only thing to worry about on this tab would be the position (and size, and background box etc.) - but hopefully these will also have some "good enough" default values. Apparently it is still kosher to say "(c) DCC" in another language/script  -  which we argued for once and were approved to do many years ago. So you've gotta trust me when I tell you that this is what it actually says: di si si  :-)
![image](https://user-images.githubusercontent.com/12962708/108127362-04455480-7110-11eb-92b4-cbbd7294c202.png)

**Comment by davidg-sil** [comment]
Seeing the image above, my first thought was 'wow, that ended up in just the right place.' I then wonder how you did that. Extra spaces, or is this from another program?

Would it be useful to be able to specify an exact horizontal position?

**Comment by mhosken** [comment]
A shift (positive or negative) along the direction of text might well be helpful. (\hskip before the text, or somesuch).

**Comment by mhosken** [comment]
In answer to the issue's question. Yes, as of the next release, credit boxes are "supposed" to work. In fact, we don't 'suppose' them to work, but we do 'hope' they work ;)

**Comment by davidc86** [comment]
:) looking forward to this. It will be a huge help.

**Comment by davidc86** [comment]
Wow! Very well done! I like how I can set up the first one like I want most of the DCC illustrations' copyright to look. Check the box to make it so, and then run PRINT.
Then I see where some overwrite the illustration itself, so for some of those I open the Details, click the button, UNCHECK the "Apply to all" box and then change the side from LEFT to RIGHT, and voila! It works, just like one would hope.
This is huge.
Much appreciated.
